### PR TITLE
[www] Adding a simple newsletter page

### PR DESCRIPTION
**Pull request by wuweiweiwu**
closes: #6236 

interactions:

initial:

screen shot 2018-07-10 at 10 05 56 pm

success:

screen shot 2018-07-10 at 10 06 21 pm

message:

screen shot 2018-07-10 at 10 06 35 pm

### CODE REVIEW

**Comment by m-allanson** [comment]
Thanks @wuweiweiwu! Preview link: https://deploy-preview-6359--gatsbyjs.netlify.com/newsletter/

cc'ing @fk for a design review and @shannonbux or @calcsam to review the text.

File: www/src/components/email-capture-form.js [hunk 1]
```diff
@@ -85,6 +85,65 @@ class EmailCaptureForm extends React.Component {
   }
 
   render() {
```
Comment (m-allanson) [inline www/src/components/email-capture-form.js]: Maybe the structure of this component would be simpler if it accepted `children`? The component can just handle the form based content, and text like:

> Enjoyed this post? Receive the next one in your inbox!

or 

> Sign up for the Gatsby newsletter.... 

can be child components. So you could use it like:

```jsx
  <EmailCaptureForm>
    <p>Enjoyed this post? Receive the next one in your inbox!</p>
  </EmailCaptureForm>
```

 See [@fk's recent PR](https://github.com/gatsbyjs/gatsby/pull/6363/files) for an example.

File: www/src/pages/newsletter.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        </Helmet>
+        <Container
+          hasSideBar={false}
+          css={{
+            display: "flex",
+            flexDirection: "column",
+          }}
+        >
+          <h1 id="introduction" style={{ marginTop: 0 }}>
+            NewsLetter
```
Comment (m-allanson) [inline www/src/pages/newsletter.js]: This 'l' should be lowercase

File: www/src/components/navigation.js [hunk 3]
```diff
@@ -159,6 +159,7 @@ export default ({ pathname }) => {
           <NavItem linkTo="/features/">Features</NavItem>
           <NavItem linkTo="/blog/">Blog</NavItem>
           <NavItem linkTo="/showcase/">Showcase</NavItem>
+          <NavItem linkTo="/newsletter/">Newsletter</NavItem>
```
Comment (m-allanson) [inline www/src/components/navigation.js]: A question for @fk - do we want to add this to the main nav? As we're fairly space constrained there.

Comment (calcsam) [reply www/src/components/navigation.js]: Unfortunately, we don't have space here -- especially with Site Showcase and Starter Showcase on the way

Comment (m-allanson) [reply www/src/components/navigation.js]: 👍 

@wuweiweiwu can you delete this line?

File: www/src/pages/newsletter.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          </h1>
+          <div>
+            Sign up for the Gatsby newsletter to keep up with the latest news
```
Comment (calcsam) [inline www/src/pages/newsletter.js]: Keep up with the latest from the Gatsby community! Hear about new features, tips & tricks, and what people are building.

File: www/src/components/email-capture-form.js (hunk 1 above)
Comment (wuweiweiwu) [reply www/src/components/email-capture-form.js]: I see. That makes a lot of sense! I'll update that component and consuming components

File: www/src/pages/newsletter.js (hunk 2 above)
Comment (wuweiweiwu) [reply www/src/pages/newsletter.js]: Oops my bad

File: www/src/components/navigation.js (hunk 3 above)
Comment (wuweiweiwu) [reply www/src/components/navigation.js]: yep!

File: www/src/pages/newsletter.js (hunk 4 above)
Comment (wuweiweiwu) [reply www/src/pages/newsletter.js]: Thanks for the blurb. That sounds much nicer

**Comment by wuweiweiwu** [comment]
@shannonbux I forgot to uncomment some code that actually did the email sign up and triggers the screen message send. I'll fix that!

**Comment by wuweiweiwu** [comment]
@m-allanson I've noticed components like [container](https://github.com/gatsbyjs/gatsby/blob/master/www/src/components/container.js) with a prop `css`. When you actually try to specify it, it conflicts with the `glamor` css attribute and the prop isnt passed properly. 

So I used `containerCss` prop instead in `email-capture-form`. Should I open an issue to fix that in the other components?

File: www/src/components/navigation.js (hunk 3 above)
Comment (fk) [reply www/src/components/navigation.js]: @m-allanson Re: adding items to the main navigation: https://github.com/gatsbyjs/gatsby/issues/6377#issuecomment-404114364 — maybe you have another idea?

**Comment by m-allanson** [comment]
> So I used containerCss prop instead in email-capture-form. Should I open an issue to fix that in the other components?

Ooh - yes please 👍

**Comment by m-allanson** [comment]
This will be published on next.gatsbyjs.org, once Gatsby v2 is live it'll be transferred over to gatsbyjs.org automatically.

**Comment by shannonbux** [comment]
Just cherry picked this

**Pull request merged**